packages/schematics/angular/app-shell/index.ts

```diff
@@ -0,0 +1,292 @@
+    }
+    const modulePath = getAppModulePath(host, app);
+    // const modulePath = getAppModulePath(host, options);
+    const moduleSource = getSourceFile(host, modulePath);
+
+    const metadataNode = getDecoratorMetadata(moduleSource, 'NgModule', '@angular/core')[0];
+    const bootstrapProperty = getMetadataProperty(metadataNode, 'bootstrap');
+
+    const arrLiteral = (<ts.PropertyAssignment> bootstrapProperty)
+      .initializer as ts.ArrayLiteralExpression;
+
+    const componentSymbol = arrLiteral.elements[0].getText();
+
+    const relativePath = getSourceNodes(moduleSource)
+      .filter(node => node.kind === ts.SyntaxKind.ImportDeclaration)
+      .filter(imp => {
+        return findNode(imp, ts.SyntaxKind.Identifier, componentSymbol);
+      })
+      .map((imp: ts.ImportDeclaration) => {
+        const pathStringLiteral = <ts.StringLiteral> imp.moduleSpecifier;
+
+        return pathStringLiteral.text;
+      })[0];
+
+    const dirEntry = host.getDir(modulePath);
+    const dir = dirEntry.parent ? dirEntry.parent.path : '/';
+    const compPath = normalize(`/${dir}/${relativePath}.ts`);
+
+    const compSource = getSourceFile(host, compPath);
+    const compMetadata = getDecoratorMetadata(compSource, 'Component', '@angular/core')[0];
+    const templateProp = getMetadataProperty(compMetadata, 'template');
+    const templateUrlProp = getMetadataProperty(compMetadata, 'templateUrl');
+
+    if (templateProp) {
+      if (!/<routerOutlet>/.test(templateProp.initializer.getText())) {
+        const recorder = host.beginUpdate(compPath);
+        recorder.insertRight(templateProp.initializer.getEnd() - 1, routerOutletMarkup);
+        host.commitUpdate(recorder);
+      }
+    } else {
+      const templateUrl = (templateUrlProp.initializer as ts.StringLiteral).text;
+      const dirEntry = host.getDir(compPath);
+      const dir = dirEntry.parent ? dirEntry.parent.path : '/';
+      const templatePath = normalize(`/${dir}/${templateUrl}`);
+      const buffer = host.read(templatePath);
+      if (buffer) {
+        const content = buffer.toString();
+        if (!/<routerOutlet>/.test(content)) {
+          const recorder = host.beginUpdate(templatePath);
+          recorder.insertRight(buffer.length, routerOutletMarkup);
+          host.commitUpdate(recorder);
+        }
+      }
+    }
+
+    return host;
+  };
+}
+
+function addServerRoutes(options: AppShellOptions): Rule {
+  return (host: Tree) => {
+    const config = getConfig(host);
+    const app = getAppFromConfig(config, options.universalApp);
+    if (app === null) {
+      throw new SchematicsException('Universal/server app not found.');
+    }
+    const modulePath = getServerModulePath(host, app);
+    if (modulePath === null) {
+      throw new SchematicsException('Universal/server app not found.');
+    }
+    let moduleSource = getSourceFile(host, modulePath);
+    if (!isImported(moduleSource, 'Routes', '@angular/router')) {
```

Comment:
If I'm reading correctly, wouldn't this cause the new shell route not to be added if `Routes` had already been imported from `@angular/router`?

Reply:
You are reading that correctly. That logic could be more granular to check for existing routes outside of the import check.

Issue created: #290